guides/source/association_basics.md

```diff
@@ -594,1804 +784,1332 @@ class Book < ApplicationRecord
-  belongs_to :author, touch: :books_updated_at
+  def title
+    content.truncate(20)
+  end
 end
 ```
 
-##### `:validate`
+This setup allows `Entry` to delegate the `title` method to its subclasses, where `Message` uses `subject` and `Comment` uses a truncated version of `content`.
 
-If you set the `:validate` option to `true`, then new associated objects will be validated whenever you save this object. By default, this is `false`: new associated objects will not be validated when this object is saved.
+Tips, Tricks, and Warnings
+--------------------------
+
+Here are a few things you should know to make efficient use of Active Record associations in your Rails applications:
 
-#### Scopes for `belongs_to`
+* Controlling caching
+* Avoiding name collisions
+* Updating the schema
+* Controlling association scope
+* Bi-directional associations
+
+### Controlling Caching
 
-There may be times when you wish to customize the query used by `belongs_to`. Such customizations can be achieved via a scope block. For example:
+All of the association methods are built around caching, which keeps the result of the most recent query available for further operations. The cache is even shared across methods. For example:
 
 ```ruby
-class Book < ApplicationRecord
-  belongs_to :author, -> { where active: true }
-end
-```
+# retrieves books from the database
+author.books.load
 
-You can use any of the standard [querying methods](active_record_querying.html) inside the scope block. The following ones are discussed below:
+# uses the cached copy of books
+author.books.size
 
-* `where`
-* `includes`
-* `readonly`
-* `select`
+# uses the cached copy of books
+author.books.empty?
+```
 
-##### `where`
+NOTE: When we use `author.books`, the data is not immediately loaded from the database. Instead, it sets up a query that will be executed when you actually try to use the data, for example, by calling methods that require data like each, size, empty?, etc. By calling `author.books.load`, before calling other methods which use the data, you explicitly trigger the query to load the data from the database immediately. This is useful if you know you will need the data and want to avoid the potential performance overhead of multiple queries being triggered as you work with the association.
 
-The `where` method lets you specify the conditions that the associated object must meet.
+But what if you want to reload the cache, because data might have been changed by some other part of the application? Just call [`reload`](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-reload) on the association:
 
 ```ruby
-class Book < ApplicationRecord
-  belongs_to :author, -> { where active: true }
-end
+# retrieves books from the database
+author.books.load
+
+# uses the cached copy of books
+author.books.size
+
+# discards the cached copy of books and goes back to the database
+author.books.reload.empty?
 ```
 
-##### `includes`
+### Counter Cache
```

Comment:
Looking at the guide as a whole, I think it makes more sense to maybe move the "Counter Cache" section under the "Detailed Associations References" section.